### DISCUSSION TITLE
Importing Teaching requests through Data Exchange

### DISCUSSION THREAD

**Discussion by kugland** [UniTime/unitime#162] [Ideas]
Is it possible to import the required instructor attributes for each course through data exchange? If it’s not, it looks like an important feature that’s missing. There is an `attributePref` element described in the DTD of the interfaces, but it looks like it's not used anywhere:

```
/unitime-master/Documentation/Interfaces% cat Preferences.dtd | grep -i attr
<!ELEMENT attributePref EMPTY >
<!ATTLIST attributePref level ( R | -2 | -1 | 0 | 1 | 2 | P ) #REQUIRED >
<!ATTLIST attributePref attribute CDATA #REQUIRED >
```

**Comment by tomas-muller** [comment]
The Preferences XML import/export currently imports/exports all preferences that can be set on classes, scheduling subparts, departments, and instructors. It does not create any teaching requests.

At the moment teaching requests and instructor attributes can be imported with the [Instructor Assignment Imports](https://github.com/UniTime/unitime/blob/master/Documentation/Scripts/Instructor%20Assignment%20Imports.xml) scripts which are described in the [Instructor Scheduling Manual](http://bit.ly/unitime-instructor-scheduling), see Data Import / Export section. These scripts can be easily modified to fit the needs of a particular institution.
